### PR TITLE
fix(richtext): narrow diff on IME text replacements to prevent span leak

### LINKED ISSUE #75: [iOS] Text attributes expand to unintended ranges during input
## Description
In the iOS sample app, there are behaviors where text formatting attributes expand to unintended ranges when typing with the iOS software keyboard.

### Issue 1: Toggled-off inline attributes are re-applied and expand backward
1. Type a word (e.g. `qwertyui`).
2. Select only a portion of the text (e.g. `ertyui`, leaving `qw` unstyled) and apply an inline attribute (e.g., text color).
3. Move the cursor to the end of the styled text.
4. Toggle the attribute **off** using the toolbar.
5. Continue typing new text (e.g. `redf`).
**Expected:** The new text should not have the attribute, and the unstyled text at the beginning (`qw`) should remain unstyled.
**Actual:** The newly typed text inherits the attribute, the toolbar toggle button turns back on, **and the unstyled text at the beginning of the word suddenly inherits the attribute as well**, coloring the entire text block.

https://github.com/user-attachments/assets/242b5910-f391-462d-931d-7a7ea1e2a329

### Issue 2: Paragraph attributes leak to new lines
1. Type a word (e.g., "heading").
2. Apply a paragraph attribute (e.g., `H1` heading).
3. Press the Return (Enter) key to start a new line.
4. Type new text on the new line.
**Expected:** Depending on the `EnterKeyStrategy`, the heading attribute should ideally be cleared on the new line (e.g., `HeadingEnterStrategy`), or act consistently with expected paragraph breaks.
**Actual:** The new line text continues to be formatted as `H1`.

https://github.com/user-attachments/assets/b67b6b99-5968-4b21-aa66-12cf9d07ed2b

## Environment
- Platform: iOS (Compose Multiplatform)
- Keyboard: iOS Software Keyboard

*(Note: Videos demonstrating these issues were provided in the related chat, please see the comments for the attached recordings).*

### PR DESCRIPTION
**Pull request by mkeeda**
Resolves #75

## Overview
This PR fixes issues where inline span attributes and paragraph attributes inadvertently leak or expand when typing with the iOS software keyboard (or any IME that relies heavily on whole-word replacements). 

## Implementation Details
- **`RichTextState.kt`**: Updated the `updateRichString` method to calculate the minimal diff (delta) from `TextFieldBuffer.changes`. By trimming common prefixes and suffixes between the original text and the new text, full-word replacements are narrowed down to precise character additions or deletions.
- This prevents spans from erroneously expanding over newly inserted text (Issue 1) and correctly triggers the `EnterKeyStrategy` when newlines are inserted via IME replacements, preventing paragraph styles from bleeding into new lines (Issue 2).
- **`RichTextStateTest.kt`**: Added comprehensive test cases to verify span behavior during text replacements and pasting.

### CODE REVIEW

File: richtext-editor/src/commonTest/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextStateTest.kt [hunk 1]
```diff
@@ -676,6 +676,146 @@ class RichTextStateTest {
...
+    @Test
+    fun `IME Return key replacement triggers EnterKeyStrategy`() {
+        val initialText = "Heading"
+        val state =
+            RichTextState(
+                initialText =
+                    RichString(text = initialText).edit {
+                        setParagraphAttribute(
+                            dev.mkeeda.arranger.richtext.HeadingKey,
+                            dev.mkeeda.arranger.richtext.HeadingLevel.H1,
```
Comment (mkeeda) [inline richtext-editor/src/commonTest/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextStateTest.kt]: プロジェクトルール（`code-style-guide.md`）に従い、`HeadingKey` と `HeadingLevel` はファイル上部でインポートしてフルパッケージ指定を避けるよう修正をお願いします（ファイル内の他数箇所についても同様にお願いします）。

**Review by mkeeda — COMMENTED** [review]
(AI review)
IMEによるテキスト置換時にSpanが意図せず拡張してしまう問題（Issue #75）に対し、`getNarrowedDelta` を用いて本当に変更があった部分の差分だけを抽出してSpanに適用する修正は、非常にロジカルで素晴らしい対応です。エッジケースに対するテストも十分に書かれており、ロジック自体はLGTMです。
*(※現在CIが実行中であるため、マージ前にはテストのパスを確認してください)*

インラインコメントで記載した、スタイル規約（フルパッケージ指定の回避）に関する修正のみご対応をお願いいたします。

File: richtext-editor/src/commonTest/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextStateTest.kt (hunk 1 above)
Comment (mkeeda) [reply richtext-editor/src/commonTest/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextStateTest.kt]: Thanks for the review! I have updated the code to use imports for `HeadingKey` and `HeadingLevel` at the top of the file, avoiding full package specifications across the entire file, as per the `code-style-guide.md`.

File: richtext-editor/src/commonTest/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextStateTest.kt [hunk 2]
```diff
@@ -676,6 +678,146 @@ class RichTextStateTest {
...
+    }
+
+    @Test
+    fun `IME Return key replacement triggers EnterKeyStrategy`() {
+        val initialText = "Heading"
+        val state =
+            RichTextState(
+                initialText =
+                    RichString(text = initialText).edit {
+                        setParagraphAttribute(
```
Comment (mkeeda) [inline richtext-editor/src/commonTest/kotlin/dev/mkeeda/arranger/richtext/editor/RichTextStateTest.kt:725]: According to the project style guide (`code-style-guide.md`), please import `HeadingKey` and `HeadingLevel` at the top of the file to avoid using full package names. (Please apply this fix to other similar occurrences in this file as well.)

**Review by mkeeda — COMMENTED** [review]
(AI review)
The fix using `getNarrowedDelta` to extract only the actual changed portion for Span application during IME text replacements (Issue #75) is very logical and excellent. The tests for edge cases are also well-written, and the logic itself LGTM.
*(Note: CI is currently running, so please ensure the tests pass before merging)*

Please address the minor style guideline issue (avoiding full package names) mentioned in the inline comment.
